### PR TITLE
do not safe log branch information

### PR DESCRIPTION
**Pull request by bavardage**
## Before this PR
It is possible to safe log branch info.

## After this PR
(per our discussion) add an errorprone check to try to prevent this.

==COMMIT_MSG==
Forbid safe logging of branch information.
==COMMIT_MSG==

### CODE REVIEW

File: baseline-error-prone/src/main/java/com/palantir/baseline/errorprone/PreventBranchSafeLogging.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        if (matcher.matches(tree, state)) {
+            if (tree.getArguments().isEmpty()) {
+                return Description.NO_MATCH;
+            }
+
+            ExpressionTree arg = tree.getArguments().get(0);
+
+            String parameterName = ASTHelpers.constValue(arg, String.class);
+
+            if (parameterName.startsWith("branch")) {
```
Comment (carterkozak) [inline baseline-error-prone/src/main/java/com/palantir/baseline/errorprone/PreventBranchSafeLogging.java:54]: @ferozco Has another check that allows a configurable set of SafeArg names to be banned, though I don't recall where it's implemented.

**Review by carterkozak — CHANGES_REQUESTED** [review]
I think this check is too specific and would prefer to use the general solution instead if we can roll that out instead. Thoughts?

**Comment by bavardage** [comment]
I was unaware of the general check - happy to make the change there if someone can point me at it

**Comment by bavardage** [comment]
closing in light of the other solution

**Comment by carterkozak** [comment]
I think it lives in another repo but I can't find it with a quick search, I'll ask the team tomorrow.

**Comment by bavardage** [comment]
yeah @ferozco shared the link - I sent an email

**Pull request closed without merging**